### PR TITLE
Automation of ST mapping process

### PR DESCRIPTION
**Pull request by anandsinghms**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/Core/STMapping/AzSKADOServiceMapping.ps1 [hunk 1]
```diff
@@ -25,11 +36,32 @@ class AzSKADOServiceMapping: CommandBase
...
+        $this.StorageRG = $env:StorageRG;
+        $this.Container = $env:Container;
+        # Report Storage settings
+        $this.ReportStorageAccount = $env:ReportStorageName;
+        $this.ReportStorageRG = $env:ReportStorageRG;
+        $this.ReportContainer = $env:ReportContainer;                
+        #get storage details
+        if($this.Auto -eq 'true'){
+            if ($this.StorageRG -and $this.StorageAccount) {
+                $keys = Get-AzStorageAccountKey -ResourceGroupName $this.StorageRG -Name $this.StorageAccount
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Core/STMapping/AzSKADOServiceMapping.ps1:50]: Add appropriate error handling

File: src/AzSK.ADO/Framework/Core/STMapping/AzSKADOServiceMapping.ps1 [hunk 2]
```diff
@@ -11,12 +11,23 @@ class AzSKADOServiceMapping: CommandBase
...
     [string] $MappingType
     [string] $OutputFolderPath
+    [string] $Auto = $false
+    [string] $StorageAccount;
+    [string] $StorageRG;
+    [string] $Container;    
+    [object] $StorageAccountCtx;     
+    # Report Storage settings
+    [string] $ReportStorageAccount;
+    [string] $ReportStorageRG;
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Core/STMapping/AzSKADOServiceMapping.ps1:21]: Please add comments for your changes
like what is storage RG and what is ReportsStorageRG, right now its not clear why they can be different

File: src/AzSK.ADO/Framework/Models/AzSKSettings.ps1 [hunk 3]
```diff
@@ -21,7 +21,7 @@ class AzSKSettings {
 	[string] $WebhookSource;
 	[string] $AutoUpdateCommand;
 	[string] $BatchScanModule;
-	[AutoUpdate] $AutoUpdateSwitch = [AutoUpdate]::On;
+	[AutoUpdate] $AutoUpdateSwitch = [AutoUpdate]::off;
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Models/AzSKSettings.ps1:24]: why is this changed required?

Comment (anandsinghms) [reply src/AzSK.ADO/Framework/Models/AzSKSettings.ps1:24]: for testing otherwise its prompt for user interaction

File: src/AzSK.ADO/Framework/Core/STMapping/AzSKADOServiceMapping.ps1 [hunk 4]
```diff
@@ -65,22 +113,27 @@ class AzSKADOServiceMapping: CommandBase
...
+            $response = Get-AzStorageBlob -Blob 'buildDefinitions.json' -Container $this.Container -Context $this.StorageAccountCtx 
+            $this.BuildSTDetails = $response.ICloudBlob.DownloadText() | ConvertFrom-Json         
+        }
+        else {
+            $this.BuildSTDetails = Get-content $this.BuildMappingsFilePath | ConvertFrom-Json    
+        }        
+        if ([Helpers]::CheckMember($this.BuildSTDetails, "data") -and ($this.BuildSTDetails.data | Measure-Object).Count -gt 0){
+            $this.BuildSTDetails.data = $this.BuildSTDetails.data | where-object {$_.ProjectName -eq $this.ProjectName}            
+            if (($this.BuildSTDetails.data | Measure-Object).Count -gt 0){
+                $this.BuildSTDetails.data[0].projectId;
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/STMapping/AzSKADOServiceMapping.ps1:128]: Should this be: $this.ProjectId = $this.BuildSTDetails.data[0].projectId

**Review by juhi037 — DISMISSED** [review]

**Review by Garima-msft — APPROVED** [review]

**Review by juhi037 — APPROVED** [review]
